cmake/modules/CheckCompiler.cmake

```diff
@@ -88,6 +88,14 @@ if (CMAKE_COMPILER_IS_GNUCXX)
   endif()
   message(STATUS "Found GCC. Major version ${GCC_MAJOR}, minor version ${GCC_MINOR}")
   set(COMPILER_VERSION gcc${GCC_MAJOR}${GCC_MINOR}${GCC_PATCH})
+  if("${GCC_MAJOR}.${GCC_MINOR}" VERSION_GREATER_EQUAL 4.9
+      AND CMAKE_GENERATOR STREQUAL "Ninja")
+    # GCC checks automatically if we are in interactive terminal mode.
+    # We use color output only for Ninja, because Ninja by default is buffering the output,
+    # so Clang disables colors as it is sure whether the output goes to a file or to a terminal.
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fdiagnostics-color=always")
+  endif()
```

Comment:
Should we do the same for clang? (-fcolor-diagnostics)?

Reply:
We did already, this PR is resolving "should we do the same for GCC as for clang" :-) See https://github.com/root-project/root/blob/cbd94894e86021288fd68cbbb330a2fb4d64198e/cmake/modules/CheckCompiler.cmake#L58

Reply:
The commit 'could' be enhanced by saying it is 'just' about gcc (since clang is already done)